### PR TITLE
Bugfix in renaming and isolation step

### PR DESCRIPTION
**Pull request by jannikseidelQBiC**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).

## Renaming before pipeline run
Other possible formats of the fastq headers was added (no special format of the header, e.g. just a number; with white spaces in the header).

## Changing the model which classifies a read in the kraken2 output as either the taxon/taxa to filter or assess 
$\frac{sumToFilter}{unclassified} \rightarrow \frac{sumToFilter}{unclassified+sumToFilter}$ is how the model changed to be compared with the `cutoff_unclassified` parameter. The description in the `nexflow_schema.json` was updated accordingly.

### CODE REVIEW

File: modules/local/isolate_ids_from_kraken2_to_blastn.nf [hunk 1]
```diff
@@ -60,14 +60,10 @@ process ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN {
...
 
-                # Add pseudo count
-                if unclassified == 0:
-                    unclassified = 1
-
                 if (
                     sum_to_filter > sum_to_keep
                     and sum_to_filter > $params.cutoff_tax2filter
-                    and sum_to_filter/unclassified > $params.cutoff_unclassified
+                    and sum_to_filter/(unclassified + sum_to_filter) > $params.cutoff_unclassified
```
Comment (d4straub) [inline modules/local/isolate_ids_from_kraken2_to_blastn.nf:66]: All fine and well, and maybe I misunderstand, but what about kmers that are neither `unclassified` nor to `sum_to_filter`, i.e. `sum_to_keep`? Isnt the total number of kmers `unclassified + sum_to_filter + sum_to_keep`?
In order to achieve a 0 to 1 ratio in the sense of `minimum fraction of kmers to filter of total kmers`, it should be `sum_to_filter/(unclassified + sum_to_filter + sum_to_keep)` ?
Therefore
```suggestion
                    and sum_to_filter/(unclassified + sum_to_filter + sum_to_keep) > $params.cutoff_unclassified
```

File: bin/rename_fastq_headers_pre.py
Comment (d4straub) [inline bin/rename_fastq_headers_pre.py:1]: I think it would be great to actually detail how valid headers look like. At least in the error message!

**Review by d4straub — COMMENTED** [review]
Looks good, but I have two comments.

File: modules/local/isolate_ids_from_kraken2_to_blastn.nf (hunk 1 above)
Comment (jannikseidelQBiC) [reply modules/local/isolate_ids_from_kraken2_to_blastn.nf:66]: The model was rewritten and is now:

$\sum(toFilter)> cutoffTax2filter$  
$\wedge$ 
$\frac{\sum(toFilter)}{\sum(toKeep)+\sum(toFilter)}> cutoffTax2keep$
$\wedge$ 
$\frac{\sum(toFilter)}{\sum(toFilter)+unclassified} > cutoffUnclassified$ 

If this evaluates to `True` the read is classified as to filter/to assess for.

File: bin/rename_fastq_headers_pre.py
Comment (jannikseidelQBiC) [reply bin/rename_fastq_headers_pre.py:1]: Was added to the script (which has been rewritten).

**Pull request merged**